interactors/nonceTransactionsHandler.go

```diff
@@ -94,6 +94,7 @@ func (nth *nonceTransactionsHandler) SendTransaction(ctx context.Context, tx *da
 	anh := nth.getOrCreateAddressNonceHandler(addressHandler)
 	if nth.checkForDuplicates && anh.isTxAlreadySent(tx) {
 		// TODO: add gas comparation logic EN-11887
+		anh.computedNonce--
```

Comment:
:+1: :+1: :+1:

Reply:
You need to protect this change with a mutex. Suggestion to create a method in the addressNonceHandler

Reply:
done